### PR TITLE
Replace type declarations

### PR DESCRIPTION
**Pull request by fingolfin**
Type declarations insert `convert` calls. Instead, use type assertions
(which just verify the object has the right type to start with), or
fix the code to actually be type stable and return the right type.

### CODE REVIEW

File: src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/methods.jl [hunk 1]
```diff
@@ -97,30 +97,30 @@ function integrate(c::CohomologyClass)::QQFieldElem
...
     end
     top_form = dict[elem]
     if iszero(top_form)
-        return 0
+        return zero(QQ)
     end
     n = AbstractAlgebra.leading_coefficient(top_form.f)
     m = AbstractAlgebra.leading_coefficient(polynomial(volume_form(toric_variety(c))).f)
-    return QQFieldElem(n//m)
+    return QQFieldElem(n,m)
```
Comment (HereAround) [inline src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/methods.jl]: @fingolfin Unless I am mistaken, this change leads to the CI-failures in 1.6 ubuntu-latest: `no method matching QQFieldElem(::QQFieldElem, ::QQFieldElem)`.

Generally speaking, your changes look good to me. In particular, I would hope that once the above line is fixed, the tests pass again.

Comment (HereAround) [reply src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/methods.jl]: Maybe `QQFieldElem(ZZ(n), ZZ(m))` might work.

Comment (HereAround) [reply src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/methods.jl]: I just tested this fix, and it seems to work. So I have taken the liberty to push a corresponding fix.

(Origin: `m`, `n` are coefficients of polynomials with rational coefficients. So by construction, they are rational numbers.)

**Review by HereAround — APPROVED** [review]
Modulo the minor change above, this PR looks good to me. Thank you @fingolfin . (But as always, let us see if the tests pass...)

**Comment by thofma** [comment]
Tests are not passing with a genuine error

File: src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/methods.jl [hunk 2]
```diff
@@ -97,30 +97,30 @@ function integrate(c::CohomologyClass)::QQFieldElem
...
     end
     top_form = dict[elem]
     if iszero(top_form)
-        return 0
+        return zero(QQ)
     end
     n = AbstractAlgebra.leading_coefficient(top_form.f)
     m = AbstractAlgebra.leading_coefficient(polynomial(volume_form(toric_variety(c))).f)
-    return QQFieldElem(n//m)
+    return QQFieldElem(ZZ(n),ZZ(m))
```
Comment (thofma) [inline src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/methods.jl]:
```suggestion
    return n//m
```

Comment (thofma) [reply src/AlgebraicGeometry/ToricVarieties/CohomologyClasses/methods.jl]: @HereAround I guess `top_form.f` and the other one are polynomials over QQ?

**Pull request merged**